### PR TITLE
Add GPGKey Implementation

### PR DESCRIPTION
**Pull request by PradyumnaKrishna**
### Description of the changes being introduced by the pull request:
A container class representing Public Key portion of the GPG is added. it consist of a `verify` method to verify the `GPGSignature` and an arbitrary data. This GPGKey will be used to verify DSSE signatures in near future.
For simplicity dataclasses are used, which initializes several methods like `__init__`, `__eq__` etc.

### Please verify and check that the pull request fulfils the following requirements:

- [x] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [x] Tests have been added for the bug fix or new feature
- [x] Docs have been added for the bug fix or new feature

cc: @lukpueh

### CODE REVIEW

File: tests/test_key.py [hunk 1]
```diff
@@ -120,6 +126,120 @@ def test_sslib_equality(self):
             self.assertEqual(sslib_key_2, sslib_key)
 
 
+@unittest.skipIf(not HAVE_GPG, "gpg not found")
+class TestGPGKey(unittest.TestCase):
```
Comment (lukpueh) [inline tests/test_key.py]: Let's combine this test case with `test_signer.TestGPGRSA` to save some redundant test code in test_key, test_signer and test_gpg.

File: securesystemslib/key.py [hunk 2]
```diff
@@ -156,3 +136,104 @@ def verify(self, signature: Signature, payload: bytes) -> bool:
...
+        }
+
+        if self.creation_time:
+            key_dict["creation_time"] = self.creation_time
+        if self.validity_period:
+            key_dict["validity_period"] = self.validity_period
+        if self.subkeys:
+            subkeys_dict = {}
+            for keyid, subkey in self.subkeys.items():
+                subkeys_dict[keyid] = subkey.to_dict()
```
Comment (lukpueh) [inline securesystemslib/key.py]: style nit: seems like a job for a dict comprehension

File: securesystemslib/key.py [hunk 3]
```diff
@@ -156,3 +136,104 @@ def verify(self, signature: Signature, payload: bytes) -> bool:
...
+            KeyError, TypeError: Invalid arguments.
+
+        """
+        subkeys_dict = key_dict.get("subkeys")
+
+        gpg_subkeys = None
+        if subkeys_dict:
+            gpg_subkeys = {}
+            for keyid, subkey_dict in subkeys_dict.items():
+                gpg_subkeys[keyid] = GPGKey.from_dict(subkey_dict)
```
Comment (lukpueh) [inline securesystemslib/key.py]: style nit: seems like a job for a dict comprehension

File: securesystemslib/key.py [hunk 4]
```diff
@@ -95,7 +75,7 @@ def from_dict(cls, key_dict: Dict[str, Any], keyid: str) -> "SSlibKey":
         keyval = key_dict.pop("keyval")
 
         # All fields left in the key_dict are unrecognized.
-        return cls(keyid, keytype, scheme, keyval, key_dict)
+        return cls(keytype, scheme, keyval, keyid, key_dict)
```
Comment (lukpueh) [inline securesystemslib/key.py:78]: Why change the argument order?

**Review by lukpueh — COMMENTED** [review]
Lovely!!

File: securesystemslib/key.py (hunk 4 above)
Comment (PradyumnaKrishna) [reply securesystemslib/key.py:78]: I thought in future `keyid` will get optional, so it's a better position for that.

File: tests/test_key.py (hunk 1 above)
Comment (PradyumnaKrishna) [reply tests/test_key.py]: you mean using `GPGKey` in `test_signer.TestGPGRSA` or something else?

Comment (lukpueh) [reply tests/test_key.py]: Yep, or keep your newly added test case and remove `test_signer.TestGPGRSA`. Either way is fine.

File: securesystemslib/key.py (hunk 4 above)
Comment (lukpueh) [reply securesystemslib/key.py:78]: Okay. Would be good to note reasoning like this in the commit message. I think we still need to have more discussion about `keyid` and the other fields, both in the keys' in-memory representation (`Key`, see https://github.com/secure-systems-lab/securesystemslib/issues/310), and in the their metadata representation (see https://github.com/secure-systems-lab/securesystemslib/issues/308). I'll make some notes in the corresponding issues.

**Review by lukpueh — APPROVED** [review]
Great work!

Just realized that `GPGSignature`, `GPGSigner` and `GPGKey` don't support `unrecognized_fields` should we add them for consistency sake with the other implementations of `Signature`, `Signer` and `Key`? I suggest to not change this PR, but would you mind creating a [downstream ticket](https://github.com/in-toto/securesystemslib/issues).

**Comment by lukpueh** [comment]
Test failure is unrelated and has been fixed upstream (kudos to @PradyumnaKrishna, see https://github.com/secure-systems-lab/securesystemslib/pull/420). Merging

**Pull request merged**